.github/actions/composite/setupNode/action.yml

```diff
@@ -29,36 +29,27 @@ runs:
       shell: bash
       run: |
         set -e
-        if [[ `git diff main --name-only | grep \.patch` != null ]]; then 
+        git fetch origin main --no-tags --depth=1
+        if git diff --name-only origin/main HEAD | grep -q '\.patch'; then
           echo 'CHANGES_IN_PATCH_FILES=true' >> "$GITHUB_OUTPUT"
-        else 
+        else
           echo 'CHANGES_IN_PATCH_FILES=false' >> "$GITHUB_OUTPUT"
         fi
 
-    - name: Patch root project node packages
-      shell: bash
-      if: | 
-        steps.patchCheck.outputs.CHANGES_IN_PATCH_FILES == 'true' && 
-        steps.cache-node-modules.outputs.cache-hit == 'true'
-      run: npx patch-package
-
-    - name: Patch node packages for desktop submodule
-      shell: bash
-      if: | 
-        steps.patchCheck.outputs.CHANGES_IN_PATCH_FILES == 'true' && 
-        steps.cache-desktop-node-modules.outputs.cache-hit == 'true'
-      run: cd desktop && npx patch-package
-
     - name: Install root project node packages
-      if: steps.cache-node-modules.outputs.cache-hit != 'true'
+      if: |
+        steps.cache-node-modules.outputs.cache-hit != 'true' ||
+        steps.patchCheck.outputs.CHANGES_IN_PATCH_FILES == 'true'
```

Comment:
I think our approach for handling diff in patch files is a bit whack. hashFiles functions works for directories, so we should just update the key from actions/cache to include a hash of the patches directory as well as node_modules. Rather than having a separate step and using git to detect the diff

Reply:
@roryabraham ??

Reply:
Subscribing

Reply:
Yes, that's what I was suggesting. The only other thing I would change from that snippet is:

```diff
- if: |
-   steps.cache-node-modules.outputs.cache-hit != 'true'
+ if: steps.cache-node-modules.outputs.cache-hit != 'true'
```

Reply:
That change should also make it so you don't have to fetch in the `patchCheck` step, because that step will be removed

Reply:
Done